[Docs] Add Install MLRun CE on Azure guide (#9926)
mlrun/mlrun: docs/install-mlrun-ce/azure-install.md

(azure-install)=
# Install MLRun CE on Azure

These instructions explain how to install MLRun CE on your Azure Kubernetes Service (AKS) cluster.

**In this section**
- [Prerequisites](#prerequisites)
- [Community Edition services](#community-edition-services)
- [Prepare values files](#prepare-values-files)
- [Create the container registry secret](#create-the-container-registry-secret)
- [Installation](#installation)
- [Usage](#usage)
- [Optional: Ingress configuration](#optional-ingress-configuration)
- [Uninstalling the chart](#uninstalling-the-chart)

## Prerequisites

> - An AKS cluster (Kubernetes >=1.34) with a default StorageClass configured

Reviewer: Instead of duplicating the Prerequisites from https://docs.mlrun.org/en/stable/install-mlrun-ce/kubernetes-install.html, let's say that all the Prerequisites from the above need to be included, and also the azure specific

Author: done

```bash
export ACR_NAME="<acr-name>"
export ACR_TOKEN=$(az acr login --name "${ACR_NAME}" --expose-token --output tsv --query accessToken)

kubectl --namespace mlrun create secret docker-registry acr-registry-credentials \
    --docker-server="<url of the registry>" \
    --docker-username="00000000-0000-0000-0000-000000000000" \
    --docker-password="${ACR_TOKEN}" \
    --docker-email="not-used@example.com"
```

The secret name must match `global.registry.secretName` in `azure-values.yaml` (default: `acr-registry-credentials`).

## Installation

1. Add the MLRun CE Helm repo:

```bash
helm repo add mlrun-ce https://mlrun.github.io/ce
helm repo update
```

2. Install MLRun CE using `azure-values.yaml`:

```bash
helm install mlrun-ce mlrun-ce/mlrun-ce \
  --namespace mlrun \
  --wait \
  --timeout 2000s \
  -f azure-values.yaml
```

## Optional: Ingress configuration

If you prefer ingress-based access instead of NodePort, optionally add `values-ingress-override-azure.yaml` to the install command. This switches exposed services to ClusterIP and enables ingress for MLRun UI, MLRun API, Nuclio, Jupyter, SeaweedFS Admin, Grafana, and Prometheus.

> 1. Download the {download}`ingress override values file template <./values-ingress-override-azure.yaml.template>`.

Reviewer: Do you think we can use this ingress template for other installations and maybe export it to https://docs.mlrun.org/en/stable/install-mlrun-ce/mlrun-ce-installation-notes.html ?

Author: moved it to installation notes and linked to it in the guide

(azure-install)=
# Install MLRun CE on Azure

> These instructions install the community edition (CE) on your Azure Kubernetes Service (AKS) cluster.

Reviewer: ```suggestion
These instructions explain how to install MLRun CE on your Azure Kubernetes Service (AKS) cluster.
```

- [Azure CLI](https://learn.microsoft.com/en-us/cli/azure/install-azure-cli) is installed and logged in (`az login`)
- [kubectl](https://kubernetes.io/docs/tasks/tools/) is configured for your AKS cluster. See [Connect to an Azure Kubernetes Service (AKS) cluster using kubectl](https://learn.microsoft.com/en-us/azure/aks/learn/quick-kubernetes-deploy-cli)
- [Helm](https://helm.sh/docs/intro/install/) version >=4.1 CLI is installed
- An accessible Docker Registry (such as [Docker Hub](https://hub.docker.com) or [Azure Container Registry (ACR)](https://learn.microsoft.com/en-us/azure/container-registry/)). The Registry's URL and credentials are consumed by the applications via a pre-created secret.
- Review the [MLRun CE installation notes](./mlrun-ce-installation-notes.md) for any additional installation steps you may need to consider.
- A bash shell to run the commands
- [Optional] If you want to store MLRun artifacts in Azure Blob Storage instead of the bundled SeaweedFS, provision an Azure Storage account and container. See [Using Azure Blob Storage for MLRun artifacts](./mlrun-ce-installation-notes.md#using-azure-blob-storage-for-mlrun-artifacts).

```{admonition} Note

> The MLRun Community Edition resources are configured initially with the default cluster/namespace resource limits. You can modify the resources from outside if needed.

Reviewer: ```suggestion
The MLRun CE resources are configured initially with the default cluster/namespace resource limits. You can modify the resources from outside if needed.
```
Also add that the user can do it by edit the `values.yaml` file - https://docs.mlrun.org/en/stable/install-mlrun-ce/mlrun-ce-installation-notes.html#advanced-chart-configuration

Author: added

```

> ## Community Edition services

Reviewer: You can remove it as it is a duplication from https://docs.mlrun.org/en/stable/install-mlrun-ce/index.html#core-components

Author: removed

The MLRun CE (Community Edition) includes the following services:
- MLRun - https://github.com/mlrun/mlrun
  - MLRun API
  - MLRun UI
  - MLRun DB (MySQL)
- Nuclio - https://github.com/nuclio/nuclio
- Jupyter - https://github.com/jupyter/notebook (+MLRun integrated)
- SeaweedFS - https://github.com/seaweedfs/seaweedfs (S3-compatible storage)
- Kafka
- TimescaleDB
- Prometheus stack - https://github.com/prometheus-community/helm-charts
  - Prometheus
  - Grafana

## Prepare values files

Create `azure-values.yaml` with your container registry and storage settings.

1. Download the {download}`azure values file template <./azure-values.yaml.template>`.
2. Copy and edit `azure-values.yaml` — fill in your registry URL, secret name, and storage settings:

```yaml
storage:
  mode: "azure-blob"
  azure:
    containerName: "data"
    accountName: "<name of the storage account>"
    connectionString: "DefaultEndpointsProtocol=https;AccountName=<name of the storage>;AccountKey=<account key>;EndpointSuffix=core.windows.net"

global:
  registry:
    url: "<url of the registry>"
    secretName: "acr-registry-credentials"
```

> ### Storage options

Reviewer: Instead of adding the example here, point to the guide in `./mlrun-ce-installation-notes.md#using-azure-blob-storage-for-mlrun-artifacts` and if needed, add the example here.
It better to have one source of truth for the examples

Author: added

`azure-values.yaml` controls where MLRun and Jupyter store artifacts:

**Option A — Azure Blob Storage**

```yaml
storage:
  mode: "azure-blob"
  azure:
    containerName: "data"
    accountName: "<name of the storage account>"
    connectionString: "<connection-string>"
```

See also [Using Azure Blob Storage for MLRun artifacts](./mlrun-ce-installation-notes.md#using-azure-blob-storage-for-mlrun-artifacts).

**Option B — Bundled SeaweedFS (default)**

> Set `storage.mode` to `"local"` to use the in-cluster SeaweedFS S3-compatible service. Kubeflow Pipelines always uses SeaweedFS regardless of this setting.

Reviewer: this is the default I think you can remove it

Author: deleted

```yaml
storage:
  mode: "local"
  local:
    accessKey: "seaweed"
    secretKey: "seaweed123"
    bucket: "mlrun"

> ```

Author: deleted the entire section

> ## Create the container registry secret

Reviewer: For all the below, please remove any duplication from the https://docs.mlrun.org/en/stable/install-mlrun-ce/kubernetes-install.html.

This page should include only commands or examples that are specific to the Azure installation.
You can remove this part 3. Verify the installation:

Author: done

kubectl --namespace mlrun create secret docker-registry acr-registry-credentials \
    --docker-server="<url of the registry>" \
    --docker-username="00000000-0000-0000-0000-000000000000" \
    --docker-password="${ACR_TOKEN}" \
    --docker-email="not-used@example.com"
```

The secret name must match `global.registry.secretName` in `azure-values.yaml` (default: `acr-registry-credentials`).

**Using Docker Hub or another private Docker registry:**

```bash
kubectl --namespace mlrun create secret docker-registry acr-registry-credentials \
    --docker-server="<registry-server>" \
    --docker-username="<registry-username>" \
    --docker-password="<registry-password>" \
    --docker-email="<user-email>"
```

## Installation

1. Add the MLRun CE Helm repo:

```bash
helm repo add mlrun-ce https://mlrun.github.io/ce
helm repo update
```

2. Install MLRun CE using `azure-values.yaml`:

```bash
helm install mlrun-ce mlrun-ce/mlrun-ce \
  --namespace mlrun \
  --wait \
  --timeout 2000s \
  -f azure-values.yaml
```

3. Verify the installation:

```bash
kubectl -n mlrun get pods
```

## Usage

> By default, MLRun CE exposes services via NodePort. Your applications are available at:

Reviewer: maybe worth adding this part to the https://docs.mlrun.org/en/stable/install-mlrun-ce/mlrun-ce-installation-notes.html and point this page to this example as it may be relevant for other installation also

Author: it exists already in https://docs.mlrun.org/en/stable/install-mlrun-ce/kubernetes-install.html#usage so I removed it

```

## Prepare values files

Create `azure-values.yaml` with your container registry and storage settings.

1. Download the {download}`azure values file template <./azure-values.yaml.template>`.
2. Copy and edit `azure-values.yaml` — fill in your registry URL, secret name, and storage settings:

```yaml
storage:
  mode: "azure-blob"
  azure:
    containerName: "data"
    accountName: "<name of the storage account>"
    connectionString: "DefaultEndpointsProtocol=https;AccountName=<name of the storage>;AccountKey=<account key>;EndpointSuffix=core.windows.net"

global:
  registry:
    url: "<url of the registry>"
    secretName: "acr-registry-credentials"
```

### Storage options

MLRun CE can store MLRun artifact data in **Azure Blob Storage** instead of the default **S3-compatible (SeaweedFS)** setup used for MLRun CE. See [Using Azure Blob Storage for MLRun artifacts](./mlrun-ce-installation-notes.md#using-azure-blob-storage-for-mlrun-artifacts).

## Create the container registry secret

**Using Azure Container Registry (ACR):**

```bash
export ACR_NAME="<acr-name>"
export ACR_TOKEN=$(az acr login --name "${ACR_NAME}" --expose-token --output tsv --query accessToken)

> kubectl --namespace mlrun create secret docker-registry acr-registry-credentials \

Reviewer 2: we take under assumption namespace was created ?

Author: Yes to not duplicate as it's already mentioned in https://docs.mlrun.org/en/stable/install-mlrun-ce/kubernetes-install.html I took example from the aws guide but if you think it's necessary I can add it
